File: src/foam/dao/LastModifiedAware.java
/**
 * @license
 * Copyright 2017 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */

package foam.dao;

import foam.mlang.predicate.Predicate;
import java.util.Date;

public interface LastModifiedAware {
  public Date getLastModified();
  public void setLastModified(Date date);
}

File: src/foam/nanos/auth/LastModifiedAware.js
/**
 * @license
 * Copyright 2017 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */

foam.INTERFACE({
  package: 'foam.nanos.auth',
  name: 'LastModifiedAware',

  properties: [
    {
      class: 'DateTime',
      name: 'lastModified',
      documentation: 'Last modified date'
    }
  ]
});
